### PR TITLE
Add support for restart, devices, capabilities.

### PR DESCRIPTION
**Pull request by zzzeek**
Adds support for the following flags:

--restart
--cap-add
--cap-remove
--device

I didn't add anything here specific to testing as it seems there's just a single integration test; to be able to test every possible combination for new flags I'd add a real unit test suite with specific JSON bits as fixtures, but wasn't sure if you're into that.

### CODE REVIEW

**Comment by assaflavie** [comment]
Thank you very much.

I'm really into testing :)

Your contribution made me get off my ass and add a bit more actual testing. If you look in `test_runlike.py` now it actually tests the correct output of various support flags.

Now, it's very incomplete, but I think you can now quite easily add coverage for your own contribution, and I'd be happy to merge it then.

**Comment by zzzeek** [comment]
i have another PR with more fixes coming as it seems the "-p" flag comes from a different part of the JSON now...

**Comment by zzzeek** [comment]
oh, they're here in this one already, NM :)

File: runlike.py [hunk 1]
```diff
@@ -40,10 +40,11 @@ def multi_option(self, path, option):
             for val in values:
                 self.options.append('--%s="%s"' % (option, val))
 
-
     def parse_ports(self):
-        ports = self.get_fact("NetworkSettings.Ports")
-        if ports is not None:
+        ports = self.get_fact("NetworkSettings.Ports") or {}
+        ports.update(self.get_fact("HostConfig.PortBindings"))
```
Comment (zzzeek) [inline runlike.py]: ugh this is wrong, if it returns None...

**Comment by assaflavie** [comment]
I've merged your code and tweaked it a bit, and added a few tests to verify. Please have a look.

If you have any idea on how to test devices on Travis, please let me know.

**Comment by assaflavie** [comment]
And thanks again for the contribution.
Closing this now. If you want to discuss the devices issue see #11

**Pull request closed without merging**